api/src/repositories/artifact-repository.ts

```diff
@@ -0,0 +1,87 @@
+import SQL from 'sql-template-strings';
+import { ApiExecuteSQLError } from '../errors/api-error';
+import { BaseRepository } from './base-repository';
+import { getLogger } from '../utils/logger';
+
+const defaultLog = getLogger('repositories/artifact-repository');
+
+export interface IArtifactMetadata {
+  title?: string;
+  description?: string;
+  file_size?: number;
+  foi_reason_description?: string;
+}
+export interface IGetArtifactMetadata extends IArtifactMetadata {
+  artifact_id: number;
+  submission_id: number;
+  uuid: string;
+  file_name: string;
+  file_type: string;
+}
+
+export class ArtifactRepository extends BaseRepository {
+  async getNextArtifactId(): Promise<number> {
+    defaultLog.debug({ label: 'getNextArtifactId' });
+
+    const sqlStatement = SQL`
+      SELECT
+        COALESCE(MAX(artifact_id), 0) + 1
+      AS
+        artifact_id
+      FROM
+        artifact;
```

Comment:
I believe there is a sequence you can call instead, and it will get the next available value for you:

```
select nextval('artifact_seq');
```